discover/discover.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package discover
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/rs/zerolog/log"
+
+	wpb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/worker"
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"google.golang.org/grpc"
+	"gopkg.in/yaml.v2"
```

Comment:
```suggestion
	"sigs.k8s.io/yaml"
```
We already use this in `vm/policy`, so better to use it here too. It claims better handling of YAML and internally uses `go-yaml` so compatible with it.